### PR TITLE
added-class-validator

### PR DESCRIPTION
**Pull request by Bharathmca68**
initially the class validators were not working i have added the package class-transformation and modified some code in main.ts file now its working perfectly  and added configuration files

### CODE REVIEW

File: .env [hunk 1]
```diff
@@ -0,0 +1,6 @@
+DB_HOST = localhost
+DB_PORT = 3309
+DB_USERNAME = root
```
Comment (Kousika1) [inline .env:3]: .env files should also be ignored and shouldn't be commited.

File: src/app.module.ts [hunk 2]
```diff
@@ -1,21 +1,23 @@
...
-      port: 3309,
-      username: 'root',
-      password: 'password',
-      database: 'user',
-      entities: [User], // provide all the entites her inside the array
-      synchronize: true,
-    }),
+    // TypeOrmModule.forRoot({
+    //   type: 'mysql',
+    //   host: 'localhost',
```
Comment (Kousika1) [inline src/app.module.ts]: Remove unused code

File: src/config/typeormconfig.ts [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+
+
+export default class TypeOrmConfig {
+    static getOrmConfig(configService: ConfigService): TypeOrmModuleOptions {
+        return {
+            type: 'mysql',
+            host: "localhost",
+            port: configService.get('DB_PORT') || 3309,
+            username: 'root',
+            password: 'password',
```
Comment (Kousika1) [inline src/config/typeormconfig.ts]: Do not commit any sensitive data like username and password. Instead, use an .env variable here

File: src/config/typeormconfig.ts [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+    imports: [ConfigModule],
+    useFactory: async (configService: ConfigService):
+        Promise<TypeOrmModuleOptions> => TypeOrmConfig.getOrmConfig(configService),
+    inject: [ConfigService]
+}
+
+////////////////////////////////////////////
+// type: 'mysql',
+    // host: 'localhost',
+    // port: 3309,
```
Comment (Kousika1) [inline src/config/typeormconfig.ts]: Remove unsed code

File: package-lock.json [hunk 5]
```diff
@@ -1432,6 +1432,26 @@
         "uuid": "8.3.2"
       }
     },
+    "@nestjs/config": {
+      "version": "1.0.1",
+      "resolved": "https://registry.npmjs.org/@nestjs/config/-/config-1.0.1.tgz",
+      "integrity": "sha512-azMl4uYlFIhYsywFxPJT81RxF3Pnn0TZW3EEmr0Wa0Wex8R2xpvBNrCcrOgW3TB1xGMP7eqBrlfsVh5ZP82szg==",
+      "requires": {
+        "dotenv": "10.0.0",
+        "dotenv-expand": "5.1.0",
```
Comment (Kousika1) [inline package-lock.json]: Add package-lock.json also in gitignore

File: src/app.module.ts (hunk 2 above)
Comment (Bharathmca68) [reply src/app.module.ts]: Removed unused codes

File: src/config/typeormconfig.ts (hunk 3 above)
Comment (Bharathmca68) [reply src/config/typeormconfig.ts]: Done

File: src/config/typeormconfig.ts (hunk 4 above)
Comment (Bharathmca68) [reply src/config/typeormconfig.ts]: Done

File: package-lock.json (hunk 5 above)
Comment (Bharathmca68) [reply package-lock.json]: added to gitignore

**Pull request merged**